modules/llrt_buffer/src/buffer.rs

```diff
@@ -332,10 +333,15 @@ impl ModuleDef for BufferModule {
         let globals = ctx.globals();
         let buf: Constructor = globals.get(stringify!(Buffer))?;
 
+        let constants = Object::new(ctx.clone())?;
+        constants.set("MAX_LENGTH", usize::MAX)?;
```

Comment:
Max length is not `usize:MAX`. I've looked this up and it is very confusing.
On my system using Node.js 22 I have 8TB as `MAX_LENGTH` = 9007199254740991 =  0x1FFFFFFFFFFFFF which seems to be derived from here from v8
https://github.com/v8/v8/blob/4c13cae8b15a7ae40007a38f8f2af7c2fdb1aa60/src/common/globals.h#L1859

However I can't allocate a Buffer (nor a typed array) that big:
```javascript
Buffer.alloc(require("buffer").constants.MAX_LENGTH)
```
```shell
Uncaught RangeError: Array buffer allocation failed
    at new ArrayBuffer (<anonymous>)
```

Reply:
I understood these constants to be the maximum possible values in the specification when there is no upper limit on computing resources.

If I want these to return the maximum possible values that can be taken in the execution environment, is there a better way to do this?

Reply:
QJS supports online u32::MAX so we can stick with this
```c
 if (len > INT32_MAX) {
        JS_ThrowRangeError(ctx, "invalid array buffer length");
        goto fail;
    }
```